src/main/java/edu/ie3/datamodel/io/source/csv/CsvIdCoordinateSource.java

```diff
@@ -53,11 +53,21 @@ public CsvIdCoordinateSource(IdCoordinateFactory factory, CsvDataSource dataSour
     this.factory = factory;
     this.dataSource = dataSource;
 
+    // validating
+    Try.ofVoid(this::validate, ValidationException.class)
+        .transformF(SourceException::new)
+        .getOrThrow();
```

Comment:
So here we start validating within the constructor, but in weather sources you remove the validation from constructor. Is there a reason for that?